### ISSUE TITLE
Keep generated features for older versions

### ISSUE THREAD

**Issue by cornzy** [ctron/package-drone#65]
At the moment the feature generator creates a new feature for every channel change. Using the version suffix ".qualifier" it is possible to have uniq versions with the current timestamp. (BTW: this should be mandatory because p2 needs to have global unique version IDs - one version means one content!)

When using the feature in your target definition you do best to manually remove the qualifier because after changing the channel your target definition breaks.

It would be nice to have the ability to keep older versions of a feature. Either keep automatically every generated featuer or give the ability to lock certain versions (I click "lock" after updating my channel).

**Comment by ctron** [comment]
I fully agree that this would be a nice feature to have. And I think it could easily be implemented by creating stored child artifacts instead of virtual. This way they would sum up and could be deleted by the cleanup aspect.

But there are a few things which have to be sorted out:

a) What if a referenced item of a feature disappears from the channel? A bundle get deleted from the channel, so the older feature would reference to a non-existing bundle.
b) The channel might get re-generated by some change, which would not change the feature itself. At the moment this it not that important, since the feature would simply be re-generated the way it was. In the new scenario this could cause a great amount of features being generated.

For b) I could imagine that we do a diff on the results and check if a new feature is required. But then again this would require access to the previous state.

**Comment by cornzy** [comment]
To a) I think it is sufficient to let the feature get invalid. Maybe display an extra warning when deleting a bundle "This bundle is already referenced in generated features".

To b) What's about manually generating of the feature? Workflow: I update the channel (maybe importing some bundles etc.) and when I'm done I click "Regenerate Feature". Otherwise the list of generated features would get quite long. Maybe optional "Generate automatically"

**Comment by ctron** [comment]
I like the idea of manually generating one. This could be added to the add artifacts menu and would allow an additional UI for selecting what is inside.
